flow/environment.py

```diff
@@ -191,6 +198,40 @@ def get_config_value(cls, key, default=flow_config._GET_CONFIG_VALUE_NONE):
         """
         return flow_config.require_config_value(key, ns=cls.__name__, default=default)
 
+    @template_filter
+    def get_prefix(cls, operation, mpi_prefix=None, cmd_prefix=None, parallel=False):
```

Comment:
I still have trouble following why we need the `mpi_prefix` and `cmd_prefix` parameters here. Do we expect that they will go away eventually? Should they maybe be the last arguments then? Do they need deprecation warnings?

Reply:
In order to not break anything we still need them here for cases where user provided `mpi_prefix` or `cmd_prefix`. I believe in the end we might want to move these to directives instead in a later release is we agree on this. So right now I only add a comment in docstring more as a note for future reference.

Reply:
Can we at least move the `parallel` argument before those two potentially obsolete arguments then? That will make it *much* easier to drop if needed.